File: src/StorekeeperAssistant.VueClient/src/models/dto/moving-dto.ts
import { MovingDetailDto } from "./moving-detail-dto";
import { WarehouseDto } from "./warehouse-dto";

export interface MovingDto {
  id: string
  movingDetails: MovingDetailDto[]
  departureWarehouse: WarehouseDto | null
  arrivalWarehouse: WarehouseDto | null
  transferDate: Date
}

File: src/Server/StorekeeperAssistant.UseCases/Movings/Queries/GetMovings/Dtos/MovingDto.cs
﻿using StorekeeperAssistant.Domain.Movings;
using System;
using System.Collections.Generic;

namespace StorekeeperAssistant.UseCases.Movings.Queries.GetMovings.Dtos;

public sealed class MovingDto
{
    public Guid Id { get; set; }
    public List<MovingDetailDto> MovingDetails { get; set; } = new List<MovingDetailDto>();
    public GetMovingsWarehouseDto? DepartureWarehouse { get; set; }
    public GetMovingsWarehouseDto? ArrivalWarehouse { get; set; }
    public DateTime TransferDate { get; set; }
    public MovementType MovementType { get; set; }
    public string MovementTypeText { get; set; } = default!;
}
